src/Microsoft.ML.Data/DataLoadSave/Binary/CodecFactory.cs

```diff
@@ -44,13 +44,13 @@ public CodecFactory(IHostEnvironment env, MemoryStreamPool memPool = null)
             _loadNameToCodecCreator = new Dictionary<string, GetCodecFromStreamDelegate>();
             _simpleCodecTypeMap = new Dictionary<DataKind, IValueCodec>();
             // Register the current codecs.
-            RegisterSimpleCodec(new UnsafeTypeCodec<DvInt1>(this));
+            RegisterSimpleCodec(new UnsafeTypeCodec<sbyte>(this));
             RegisterSimpleCodec(new UnsafeTypeCodec<byte>(this));
-            RegisterSimpleCodec(new UnsafeTypeCodec<DvInt2>(this));
+            RegisterSimpleCodec(new UnsafeTypeCodec<short>(this));
             RegisterSimpleCodec(new UnsafeTypeCodec<ushort>(this));
-            RegisterSimpleCodec(new UnsafeTypeCodec<DvInt4>(this));
```

Comment:
I have a test that ensures it will succeed. 

---
In reply to: [212768406](https://github.com/dotnet/machinelearning/pull/683#discussion_r212768406) [](ancestors = 212768406)